### PR TITLE
Fixes bug: RealTime price not set every second

### PR DESCRIPTION
**Pull request by AlexCatarino**
On LiveTradingResultHandler.ProcessSynchronousEvents, realtime price was assigned to Security.Cache._lastData.Value and that assignment did not change Security.Price used to update Portfolio statistics (eg. TotalPortfolioValue).
For Forex securities, adds Portfolio.CashBook updates with realtime price.

### CODE REVIEW

File: Common/Securities/SecurityCache.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public class SecurityCache
         /// <summary>
         /// Gets the most recent price submitted to this cache
         /// </summary>
-        public decimal Price { get; private set; }
+        public decimal Price { get; set; }
```
Comment (mchandschuh) [inline Common/Securities/SecurityCache.cs]: I think there is a set market price function on security that should be called to do this. If a property is private set, care should be taken when changing to public.

File: Engine/Results/LiveTradingResultHandler.cs [hunk 2]
```diff
@@ -840,29 +841,30 @@ public void StoreResult(Packet packet, bool async = true)
 
                         var highResolutionCharts = new Dictionary<string, Chart>(live.Results.Charts);
 
-                        // 10 minute resolution data, save today
-                        var tenminuteSampler = new SeriesSampler(TimeSpan.FromMinutes(10));
-                        var tenminuteCharts = tenminuteSampler.SampleCharts(live.Results.Charts, start, stop);
+                        // 1 minute resoluton data, save today
+                        var minuteSampler = new SeriesSampler(TimeSpan.FromMinutes(1));
+                        var minuteCharts = minuteSampler.SampleCharts(live.Results.Charts, start, stop);
```
Comment (mchandschuh) [inline Engine/Results/LiveTradingResultHandler.cs]: Are these changes intentional? They seem unrelated to the PR title.

File: Engine/Results/LiveTradingResultHandler.cs [hunk 3]
```diff
@@ -840,29 +841,30 @@ public void StoreResult(Packet packet, bool async = true)
...
-                        live.Results.Charts = minuteCharts;
+                        live.Results.Charts = tenminuteCharts;
                         data_keys.Add(new
                         {
-                            Key = CreateKey("minute"),
+                            Key = CreateKey("10minute"),
                             Serialized = JsonConvert.SerializeObject(live.Results)
                         });
-
+
```
Comment (mchandschuh) [inline Engine/Results/LiveTradingResultHandler.cs]: Excess white space

File: Engine/Results/LiveTradingResultHandler.cs [hunk 4]
```diff
@@ -1021,17 +1022,26 @@ public void ProcessSynchronousEvents(bool forceProcess = false)
...
+                                    cash.Update(last);
+                                }
                             }
                             else
                             {
                                 // we haven't gotten data yet so just spoof a tick to push through the system to start with
                                 security.SetMarketPrice(new Tick(DateTime.Now, subscription.Configuration.Symbol, price, price));
                             }
-
+
```
Comment (mchandschuh) [inline Engine/Results/LiveTradingResultHandler.cs]: Excess white space

File: Engine/Results/LiveTradingResultHandler.cs [hunk 5]
```diff
@@ -1021,17 +1022,26 @@ public void ProcessSynchronousEvents(bool forceProcess = false)
...
                             {
                                 // we haven't gotten data yet so just spoof a tick to push through the system to start with
                                 security.SetMarketPrice(new Tick(DateTime.Now, subscription.Configuration.Symbol, price, price));
                             }
-
+                            
                             //Sample Asset Pricing:
                             SampleAssetPrices(subscription.Configuration.Symbol, time, price);
-                        }
+                        }
```
Comment (mchandschuh) [inline Engine/Results/LiveTradingResultHandler.cs]: Excess white space

File: Engine/Results/LiveTradingResultHandler.cs (hunk 2 above)
Comment (AlexCatarino) [inline Engine/Results/LiveTradingResultHandler.cs]: They are. I am going to undo them and delete white spaces.

File: Engine/Results/LiveTradingResultHandler.cs [hunk 6]
```diff
@@ -1021,17 +1022,26 @@ public void ProcessSynchronousEvents(bool forceProcess = false)
                             var last = security.GetLastData();
                             if (last != null)
                             {
-                                last.Value = price;
+                                security.Cache.Price = price;
```
Comment (mchandschuh) [inline Engine/Results/LiveTradingResultHandler.cs]: We can pump this new last value back into set market price. My fear is having two methods of doing this, but one updates other things, like models.

File: Common/Securities/SecurityCache.cs (hunk 1 above)
Comment (jaredbroad) [inline Common/Securities/SecurityCache.cs]: SetMarketPrice is updating volatility models so would cause different data in backtest/live; but might be good to have a SetRealTimePrice() setter method rather than setting the property directly

Comment (AlexCatarino) [inline Common/Securities/SecurityCache.cs]: Ok. Let's do the SetRealTimePrice() option.
